resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverBuilder.java

```diff
@@ -55,13 +66,68 @@
     private int maxPayloadSize = 4096;
     private boolean optResourceEnabled = true;
     private HostsFileEntriesResolver hostsFileEntriesResolver = HostsFileEntriesResolver.DEFAULT;
-    private DnsServerAddressStreamProvider dnsServerAddressStreamProvider = platformDefault();
+    private DnsServerAddressStreamProvider dnsServerAddressStreamProvider = defaultStreamProvider();
     private DnsQueryLifecycleObserverFactory dnsQueryLifecycleObserverFactory =
             NoopDnsQueryLifecycleObserverFactory.INSTANCE;
     private String[] searchDomains;
     private int ndots = -1;
     private boolean decodeIdn = true;
 
+    static {
+        Constructor<? extends DnsServerAddressStreamProvider> constructor = null;
+        if (PlatformDependent.isOsx()) {
```

Comment:
In `DnsServerAddressStreamProviders` you decide between windows and unix. Consider moving all this macos logic there as well to keep all os-related decision in one place.

Reply:
let me see what I can do.. `MacOSDnsServerAddressStreamProvider` does call `DnsServerAddressStreamProviders.platformDefault()` by itself so this would create a cycle..

Reply:
@idelpivnitskiy done... PTAL again... also @NiteshKant